### PR TITLE
Be more accepting in UnboundedReadDeduplicatorTest

### PR DESCRIPTION
**Pull request by tgroh**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [x] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [x] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [x] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [x] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.txt).

---

Don't depend on all the threads failing. Instead, assert that at most
one success was encountered, and we saw at most numThreads - 1 failures.

### CODE REVIEW

**Comment by tgroh** [comment]
R: @dhalperi

File: runners/direct-java/src/test/java/org/apache/beam/runners/direct/UnboundedReadDeduplicatorTest.java [hunk 1]
```diff
@@ -83,7 +84,7 @@ public void cachedIdDeduplicatorMultithreaded() throws InterruptedException {
     executor.shutdownNow();
 
     assertThat(successCount.get(), equalTo(1));
-    assertThat(failureCount.get(), equalTo(numThreads - 1));
+    assertThat(failureCount.get(), lessThan(numThreads));
```
Comment (dhalperi) [inline runners/direct-java/src/test/java/org/apache/beam/runners/direct/UnboundedReadDeduplicatorTest.java]: add comment explaining why this semantic change is appropriate.

Comment (tgroh) [inline runners/direct-java/src/test/java/org/apache/beam/runners/direct/UnboundedReadDeduplicatorTest.java]: Done.

**Comment by dhalperi** [comment]
LGTM. Will merge when green, or harass any committer if I end up getting sidetracked

**Pull request closed without merging**